docs/samples/metrics-and-monitoring/README.md

```diff
@@ -0,0 +1,76 @@
+# Metrics and Monitoring
+
+> Getting started with Prometheus-based monitoring of KFServing models.
+
+# Table of Contents
+1. [Install Prometheus](#install-prometheus)
+2. [Access Prometheus Metrics](#access-prometheus-metrics)
+3. [Metrics-driven experiments and progressive delivery](#metrics-driven-experiments-and-progressive-delivery)
+4. [Removal](#removal)
+
+## Install Prometheus
+
+**Prerequisites:** Kubernetes cluster and [Kustomize v3](https://kubectl.docs.kubernetes.io/installation/kustomize/).
+
+Install Prometheus using Prometheus Operator.
+
+```shell
+cd kfserving
+kustomize build docs/samples/metrics-and-monitoring/prometheus-operator | kubectl apply -f -
+kubectl wait --for condition=established --timeout=120s crd/prometheuses.monitoring.coreos.com
+kubectl wait --for condition=established --timeout=120s crd/servicemonitors.monitoring.coreos.com
+kustomize build docs/samples/metrics-and-monitoring/prometheus | kubectl apply -f -
+```
+
+## Access Prometheus Metrics
+In this section, we will use a v1beta1 InferenceService sample to demonstrate how to access Prometheus metrics that are automatically generated by [Knative's queue-proxy container](https://knative.dev) for your KFServing models.
+
+1. `kubectl create ns kfserving-test`
+2. `cd docs/samples/v1beta1/sklearn`
+3. `kubectl apply -f sklearn.yaml -n kfserving-test`
+4. If you are using a Minikube based cluster, then in a separate terminal, run `minikube tunnel` and supply password if prompted.
+5. In a separate terminal, follow [these instructions](https://github.com/kubeflow/kfserving/blob/master/README.md#determine-the-ingress-ip-and-ports) to find and set your ingress IP, host, and service hostname. Then, send prediction requests to the `sklearn-iris` model you created in Step 3. above as follows.
+```
+while clear; do \
+  curl -v \
+  -H "Host: ${SERVICE_HOSTNAME}" \
+  -d @./iris-input.json \
+  http://${INGRESS_HOST}:${INGRESS_PORT}/v2/models/sklearn-iris/infer
+  sleep 0.3
+done
+```
+6. In a separate terminal, port forward the Prometheus service.
+```shell
+kubectl port-forward service/prometheus-operated -n kfserving-monitoring 9090:9090
+```
+7. Access Prometheus UI in your browser at http://localhost:9090
```

Comment:
```suggestion
7. Access Prometheus UI in your browser at http://localhost:9090 or http://${INGRESS_HOST}:9090
```

Reply:
Access Prometheus through Istio-ingress actually requires reconfiguring the Istio ingress gateway. By default, Istio-ingress is not set up to expose Prometheus service externally (please see comment above).